src/kudu/tserver/simple_tablet_manager.cc

```diff
@@ -390,8 +390,13 @@ Status TSTabletManager::Start(bool is_first_run) {
   // The existing binlog OpId's might have no bearing
   // on the new ring. We will push the problem of figuring out catchup
   // and joining the ring to AddInstance implementation.
-  if (!is_first_run && server_->opts().log_factory) {
+  if (server_->opts().log_factory && (!is_first_run
+      || server_->opts().log_bootstrap_on_first_run)) {
     log_->GetRecoveryInfo(&bootstrap_info);
+    if (is_first_run && server_->opts().log_bootstrap_on_first_run &&
+        bootstrap_info.last_id.term() > consensus_->CurrentTerm()) {
```

Comment:
Should we do this check here? `SetCurrentTermBootstrap` is already doing this and I think we should fail the Init if this happens.

Reply:
It has to do with how SetCurrentTermBootstrap is implemented. SetCurrentTermBootstrap is implemented like "SetCurrentTerrmUnlocked" and is expected to be called only when we incrrease the current term of the instance up to a higher logged term. It returns bad status if it is called with a lower or equal term in the log.

However having lower or equal term in the log are fine. An entity can have lower logged term if it got an AppendEntries, then timed out and bumped its term for an election and then crashed. Having an equal term as current term is also the common case. However if we call SetCurrentTermBootstrap blindly by using bootstrap_info.last_id.term() , it will return error.